app/src/main/java/org/fossasia/susi/ai/dataclasses/ReportSkill.kt

```diff
@@ -0,0 +1,13 @@
+package org.fossasia.susi.ai.dataclasses
+
+/**
+ * @author : codedsun
+ * Created on 04/09/18
+ */
+data class ReportSkill(
+        val model: String,
+        val group: String,
+        val skill: String,
+        val feedback: String,
+        val accessToken: String
```

Comment:
Is the key name for access token same as the variable name used here?

Reply:
yes!

Reply:
```
/**
      * Report a skill
      * @param query A query  map consisting of the following key value pairs
      *              model       Model of the skill(Eg. general)
      *              group       Group of the skill(Eg. Knowledge)
      *              skill       Skill to be reported
      *              feedback    The report/msg send by the user
      *              accessToken access token of the user
      * @return the call
```

Reply:
val map: MutableMap<String, String> = HashMap()
        map.put("access_token", query.accessToken)
        map.put("feedback", query.feedback)
        map.put("skill", query.skill)
        map.put("group", query.group)
        map.put("model", query.model)

This is the map which goes as query into retrofit instance

Reply:
I was not asking for this. This is all the code that you have been writing. Also, note - Here also, the key is "access_token". Please log the response that is received when you flag a skill as inappropriate. Then check that response, you'll see all the keys clearly.

Reply:
This is the type of response that we get
package org.fossasia.susi.ai.rest.responses.susi

/**
 * Response received on reporting a skill successfully
 *
 * @author : codedsun
 * Created on 04/09/18
 */
data class ReportSkillResponse(
        val feedback: String,
        val accepted: String,
        val message: String)


What's the issue we are facing?

Reply:
@arundhati24 I have checked that earlier, and tested this successfully where we are stuck? and what's the issue that we are facing?

Reply:
The issue is that you are not using the correct key name. According to me, the key for access token is "access_token" and for that you need to use serialize it.
But, to confirm, you should log the response and check it.

Reply:
Okay @arundhati24 the params in the URL is 
```
 http://localhost:4000/cms/reportSkill.json?model=general&group=Knowledge&skill=some_inappropriate_skill&feedback=report_message&access_token=6O7cqoMbzlClxPwg1is31Tz5pjVwo3. 
```

but why do we need to change that key in data class, we are just using this as setter and getter and in SkillDetailsModel class we are already pushing the data according to the params that are accepted and their it's clearly mentioned 'access_token' as the key.

Reply:
We have also done the same in PostFeedback class as the key their mentioned is not same as access_token that's used in the URL.

Tell me what changes should I do ?

Reply:
Okay @codedsun I checked it. This is the query object and not the response. I took it for the response data class. To avoid any such confusion in future, could you please change the name of the data class to ReportSkillQuery?

Reply:
Yes sure!